### PR TITLE
Improve README titles and add TOC to samples README

### PR DESCRIPTION
**Pull request by bobwatso-aws**
*Issue #, if available:* N/A
*Description of changes:* Minor text-only edits to provide a smoother transition from the AWS IoT Core Developer Guide to the SDK documentation. (e.g. the links in https://docs.aws.amazon.com/iot/latest/developerguide/iot-connect-devices.html#iot-connect-device-sdks that land on an SDK README page.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samples/README.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
+# Sample apps for the AWS IoT Device SDK for Embedded C
+
+All samples are written in C unless otherwise mentioned. The these sample apps are included in the SDK and described below.
+ * [`subscribe_publish_sample`](0#subscribe-publish-sample) - a simple pub/sub MQTT example
```
[the remark this answers is not in the document]
Comment (bobwatso-aws) [reply samples/README.md]: It is indeed. Thanks for catching that.
The more recent commit fixes it.

**Review by muneebahmed10 — APPROVED** [review]

File: samples/README.md [hunk 2]
```diff
@@ -1,3 +1,15 @@
+# Sample apps for the AWS IoT Device SDK for Embedded C
+
+All samples are written in C unless otherwise mentioned. The these sample apps are included in the SDK and described below.
+ * [`subscribe_publish_sample`](#subscribe-publish-sample) - a simple pub/sub MQTT example
+ * [`subscribe_publish_cpp_sample`](#subscribe-publish-cpp-sample) - a simple pub/sub MQTT example written in C++
```
Comment (aggarg) [inline samples/README.md:5]: Is there a CPP sample - I do not see it. Even the Makefile does not have anything related to C++: https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/master/samples/linux/subscribe_publish_sample/Makefile

File: samples/README.md [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+ * [`subscribe_publish_library_sample`](#subscribe-publish-library-sample) - a simple pub/sub MQTT example which builds the SDK as a separate library
+
+ These sample apps are also provided in this SDK.
+ * [`shadow_sample`](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/master/samples/linux/shadow_sample) - a simple device shadow example using a connected window example
+ * [`shadow_sample_console_echo`](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/master/samples/linux/shadow_sample_console_echo) - a sample to work with the AWS IoT Console interactive guide
+ * [`jobs_sample`](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/master/samples/linux/jobs_sample) - Establishes a connection to the AWS IoT MQTT Platform and performs several operations to demonstrate the basic capabilities of the AWS IoT Jobs platform.
```
Comment (aggarg) [inline samples/README.md:11]: How about the following to be consistent with the above:

```
A sample to demonstrate the capabilities of the AWS IoT Jobs service
```

Please ensure either all the sample descriptions end with period or none does.

File: samples/README.md (hunk 2 above)
Comment (bobwatso-aws) [reply samples/README.md]: Removed references to non-existent sample.

File: samples/README.md (hunk 3 above)
Comment (bobwatso-aws) [reply samples/README.md]: Edited all descriptions for consistency.

**Review by aggarg — APPROVED** [review]

**Review by muneebahmed10 — APPROVED** [review]

**Pull request merged**